libs/language-server/src/lib/ast/expressions/evaluators/replace-operator-evaluator.ts

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// eslint-disable-next-line import/no-cycle
+import { DefaultTernaryOperatorEvaluator } from '../operator-evaluator';
+import { REGEXP_TYPEGUARD, STRING_TYPEGUARD } from '../typeguards';
+
+export class ReplaceOperatorEvaluator extends DefaultTernaryOperatorEvaluator<
+  string,
+  RegExp,
```

Comment:
This would be stronger as Regex OR string (so you could do `"Example" replace "ample" with "sample"` which is easier for non-programmers because you do not need to know regexes). I would like to add that in a follow up task if it is possible to do.

Reply:
I think we would have to extend the `InternalValueRepresentation` to support type unions, then it should work.